### PR TITLE
Add on_event as requested in issue #58

### PR DESCRIPTION
**Pull request by pbsds**
Tested with:

```python
#!/usr/bin/env python3
import responder
import logging

root = logging.getLogger().setLevel(logging.DEBUG)

api = responder.API()

@api.route("/{greeting}")
async def greet_world(req, resp, *, greeting):
    resp.text = f"{greeting}, world!"

@api.on_event('startup')
async def asd():
    print("startup")

@api.on_event('tick', seconds=5)
async def asd():
    print("tick")

@api.on_event('cleanup')
async def asd():
    print("cleanup")

if __name__ == '__main__':
    api.run()#log_level="debug")
```

Which gave the output:

```shell
pbsds@bertil responder (master)$ pipenv run ./test.py 
INFO: Started server process [30447]
INFO: Waiting for application startup.
startup
INFO: Uvicorn running on http://127.0.0.1:5042 (Press CTRL+C to quit)
tick
tick
tick
^CINFO: Stopping server process [30447]
INFO: Waiting for application cleanup.
cleanup
```

### CODE REVIEW

File: responder/api.py [hunk 1]
```diff
@@ -300,6 +305,30 @@ def no_response(req, resp, **params):
 
         return resp
 
+    def add_event_handler(self, event_type, handler, seconds=None):
+        """Add a event handler to the API.
+
+        :param event_type: A string in ("startup", "cleanup", "shutdown", "tick")
```
Comment (tomchristie) [inline responder/api.py]: Just use “startup” and “shutdown”, not “cleanup”.

File: responder/api.py [hunk 2]
```diff
@@ -300,6 +305,30 @@ def no_response(req, resp, **params):
...
+        :param handler: The function to run. Can be either a function or a coroutine
+        :param seconds: Used when event_type=="tick", as time between each tick
+        """
+
+        if event_type.lower() == "tick":
+            assert seconds != None
+            async def ticker(handler, seconds):
```
Comment (tomchristie) [inline responder/api.py]: Couple of things here: really you need to co-ordinate timer events to stop properly on shutdown. (And only complete shutdown once any pending timer event completes.) Also the sync calls ought to happen within a threadpool (there’s other places where this needs to happen too)

I’d consider perhaps raising a PR just for startup/shutdown first, then looking at timer events separately.

(I’ll be implementing the same for Starlette next week, which might be helpful here.)

**Comment by kennethreitz** [comment]
I think I'd like to stay away from this, for now.

**Comment by pbsds** [comment]
Why? I understand dropping the tick event, but startup and shutdown events are kinda neccesary for most people wanting to use a dbpool or creating tables for their APIs/services. What _would_ you like?

This is the second closed PR (see #105) where _I'm not getting a lot of feedback to work with_.
I like what i see in responder so far, and would like to switch over to using it myself, which is why i'm contributing. But this is kinda demotivating.

Are you perhaps interested in a PR with the only first commit (without the tick event)? Or should i add the thread pool? Do you want to see a totally different approach in implementation?

**Comment by kennethreitz** [comment]
thanks!

**Pull request merged**